src/main/java/appeng/crafting/v2/CraftingCalculations.java

```diff
@@ -59,32 +57,73 @@ public static <StackType extends IAEStack<StackType>> void registerByteAmountAdj
     public static <StackType extends IAEStack<StackType>> List<CraftingTask> tryResolveCraftingRequest(
             CraftingRequest<StackType> request, CraftingContext context) {
         final ArrayList<CraftingTask> allTasks = new ArrayList<>(4);
-        for (final CraftingRequestResolver<?> unsafeProvider : Multimaps
-                .filterKeys(providers, key -> key.isAssignableFrom(request.stackTypeClass)).values()) {
-            try {
-                // Safety: Filtered by type using Multimaps.filterKeys
-                final CraftingRequestResolver<StackType> provider = (CraftingRequestResolver<StackType>) unsafeProvider;
-                final List<CraftingTask> tasks = provider.provideCraftingRequestResolvers(request, context);
-                allTasks.addAll(tasks);
-            } catch (Exception t) {
-                AELog.log(
-                        Level.WARN,
-                        t,
-                        "Error encountered when trying to generate the list of CraftingTasks for crafting {}",
-                        request.toString());
+
+        Class<? extends IAEStack<?>>[] keyArray = providers.keyArray();
```

Comment:
nit: I like using final to indicate something shouldn't be re-written, and in some cases it makes the JIT optimizer smarter (but not usually)

Reply:
It doesn't matter for the JIT since it's not present in bytecode. I'll change to keep it consistent with the rest of the code.